provider/digest_provider.py

```
@@ -199,6 +199,7 @@ def put_digest_to_endpoint(logger, digest_id, digest_content, settings):
         logger.exception(
             "Exception issuing PUT to the digest endpoint for digest_id %s. Details: %s" %
             (str(digest_id), str(exception)))
+        raise
```

Does the exception need to be re-raised here, or would it not flow through? No harm I guess

---

Yes, unless Python has some specific behavior I don't remember, re-raising must be explicit as otherwise `except` swallows the exception